### ISSUE TITLE
Access to HeightTree information and events when it changes.

### ISSUE THREAD

**Issue by Ttxman** [icsharpcode/AvalonEdit#358]
For our internal tool I have added VS like markings to scrollbar for search results and spellcheck (No dark theme for now..)
![image](https://user-images.githubusercontent.com/2779184/181279616-17c33fdb-d999-442f-979e-626be28c7764.png)
(I have public [project](https://github.com/Ttxman/WpfRangeControls) with WPF to do the marks)

I needed to slightly [modify](https://github.com/Ttxman/AvalonEdit/commit/fb86f61f3e23211aee332935136ba94a82bfb6b1) to AvalonEdit source code and make some information about height tree public. 
Making internal classes public does not feel like a right way to do it and I don't know AvalonEdit internals good enough to make it clean enough for pull request.

It would be really nice to have some official way released in AvalonEdit nuget to do these :)

**Comment by siegfriedpammer** [comment]
I am not exactly sure why you would need access to the height tree to render symbols for text markers on a scroll bar based on their location in the document. SharpDevelop has a fairly simple reusable implementation.

See https://github.com/icsharpcode/SharpDevelop/blob/master/src/AddIns/DisplayBindings/AvalonEdit.AddIn/Src/EnhancedScrollBar.cs

Let me know what you think!

**Comment by Ttxman** [comment]
@siegfriedpammer TY This is much better solution.

When I tried to align marking just by line numbers it obviously didn't work because of word wrap. Then I tried to use `TextView.GetVisualPosition()` which was rerendering all lines.

For some reason I entirely missed `TextView.GetVisualTopByDocumentLine()` and took deep dive on how scrolling and visual line caching works, which led me to HeightTree. So in the end I have practically just reimplemented GetVisualTopByDocumentLine by directly accessing HeightTree. :/
